src/index.js

```diff
@@ -205,3 +217,10 @@ export function CartProvider({
     </CartContext.Provider>
   );
 }
+
+CartProvider.propTypes = propTypes
+
+CartProvider.defaultProps = {
+  id : Math.random().toString(36).substring(7),
```

Comment:
We could move this up above when destructuring `CartProvider` props.

Reply:
Of course. no need of proptypes also. Just give this default value when destructuring